### PR TITLE
Problem: omni_vfs backend support is still limited 

### LINKED ISSUE #402: Problem: omni_vfs backend support is still limited
Currently, omni_vfs only has only one backend: `local_fs`. This is very limiting as this is not as practical on the server.

Proposed solution: implement a `table_fs` that will provide the same API for table-backed filesystems.

Different designs can be considered, but it is important to ensure the performance of finding/listing files under a directory (recursively) as well as direct lookup.

I've had a (not a correctly implemented) attempt before: https://github.com/yrashk/omnigres/commit/b159f5631d84946af574c3b4faa5100b1b9001f2. Feel free to consult with it.

---

**Bounty size: $250**

[Read more about bounties](https://github.com/omnigres/omnigres/wiki/Bounties)

### PR DESCRIPTION
**Pull request by rmodpur**
Currently, omni_vfs only has only one backend: local_fs.
This is very limiting as this is not as practical on the server.

Solution: implement a table_fs that will provide the same API for table-backed filesystems

Closes #402

### CODE REVIEW

**Comment by rmodpur** [comment]
@yrashk This approach looks promising
I have only implemented insert and list bit till now, but i inserted about 1.2M files mimicking my home directory structure. I was able to list files in a directory in about 440 ms

**Comment by rmodpur** [comment]
@yrashk changes are ready to be reviewed, listing of directory takes less than 150ms when filesystem has more than million files.

**Comment by yrashk** [comment]
This PR effectively superseded #403, correct?

File: extensions/omni_vfs/migrate/4_table_fs.sql [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+pg_major_version int;
+begin
+    select
+        split_part(setting, '.', 1)
+    from pg_settings
+    where
+        name = 'server_version'
+    into pg_major_version;
+    
+    if pg_major_version >= 15  then
```
[the remark this answers is not in the document]
Comment (rmodpur) [reply extensions/omni_vfs/migrate/4_table_fs.sql:29]: Yes

**Comment by rmodpur** [comment]
> This PR effectively superseded #403, correct?

Correct

File: extensions/omni_vfs/migrate/4_table_fs.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmodpur) [reply extensions/omni_vfs/migrate/4_table_fs.sql]: I have scraped the partial index commit and replaced it with better method, now the performance is consistent across all pg versions

**Comment by yrashk** [comment]
I've updated the title and the description of this PR to match the original description prior to merging.

**Pull request merged**